packages/react-components/src/organisms/EngagementSort.tsx

```diff
@@ -0,0 +1,145 @@
+import { css, Theme } from '@emotion/react';
+import { useEffect, useRef, useState } from 'react';
+
+import {
+  paper,
+  steel,
+  colorWithTransparency,
+  tin,
+  pine,
+  mint,
+  neutral900,
+} from '../colors';
+import { GeneralSortingIcon } from '../icons';
+import { rem } from '../pixels';
+
+const sortButton = css({
+  background: 'none',
+  border: 'none',
+  cursor: 'pointer',
+});
+
+const menuContainer = css({
+  position: 'relative',
+});
+
+const menuStyles = css({
+  position: 'absolute',
+  width: rem(233),
+  right: `-${rem(6)}`,
+  top: `-${rem(6)}`,
+
+  backgroundColor: paper.rgb,
+  border: `1px solid ${steel.rgb}`,
+  boxShadow: `0 2px 6px 0 ${colorWithTransparency(tin, 0.34).rgba}`,
+
+  display: 'none',
+  flexDirection: 'column',
+
+  padding: `${rem(8)} 0`,
+  boxSizing: 'border-box',
+});
+
+const showMenuStyles = css({
+  display: 'flex',
+});
+
+const optionStyles = css({
+  cursor: 'pointer',
+
+  display: 'flex',
+  alignItems: 'center',
+  gap: rem(8),
+  padding: `${rem(8)} ${rem(16)}`,
+
+  color: neutral900.rgba,
+  border: 'none',
+  backgroundColor: 'unset',
+
+  ':hover': {
+    backgroundColor: mint.rgba,
+    color: pine.rgba,
+  },
+});
+
+const hoverStyles = ({
```

Comment:
I've added to GP2 locally to test it and it worked 🎉 
<img width="449" alt="Screenshot 2024-08-01 at 12 30 32" src="https://github.com/user-attachments/assets/b16627e2-04d7-449d-8905-317ed38ce665">